qiita_db/metadata_template/util.py

```diff
@@ -315,3 +331,115 @@ def get_invalid_sample_names(sample_names):
             inv.append(s)
 
     return inv
+
+
+def looks_like_qiime_mapping_file(fp):
+    """Checks if the file looks like a QIIME mapping file
+
+    Parameters
+    ----------
+    fp : str or file-like object
+        filepath to check if it looks like a QIIME mapping file
+
+    Returns
+    -------
+    bool
+        True if fp looks like a QIIME mapping file, false otherwise.
+
+    Raises
+    ------
+    QiitaDBError
+        If an empty file is passed
+
+    Notes
+    -----
+    This is not doing a validation of the QIIME mapping file. It simply checks
+    the first line in the file and it returns true if the line starts with
+    '#SampleID', since a sample/prep template will start with 'sample_name' or
+    some other different column.
+    """
+    first_line = None
+    with open_file(fp, mode='U') as f:
+        first_line = f.readline()
+    if not first_line:
+        raise QiitaDBError('Empty file passed!')
```

Comment:
That is actually a very good point. An empty file does not look like a QIIME mapping file, so the right thing to do is return False. The process will raise in another step though, but this will reduce the amount of code duplication that we have.